src/client/clientmap.cpp

```diff
@@ -677,12 +677,24 @@ void ClientMap::renderPostFx(CameraMode cam_mode)
 	MapNode n = getNode(floatToInt(m_camera_position, BS));
 
 	const ContentFeatures& features = m_nodedef->get(n);
+
+	// Account for the near clipping plane to counter see-through "glitches":
+	// Check all nodes within a radius of near_plane (Manhattan distance)
+	bool any_nearby_solid = false;
+	f32 near_plane = m_client->getCamera()->getCameraNode()->getNearValue();
+	for (s8 x = -1; x <= 1; x += 2) {
+		for (s8 y = -1; y <= 1; y += 2) {
+			for (s8 z = -1; z <= 1; z += 2) {
+				any_nearby_solid |= m_nodedef->get(getNode(floatToInt(
+						m_camera_position + near_plane * v3f(x, y, z), BS))).solidness == 2;
+			}
+		}
+	}
+
 	video::SColor post_effect_color = features.post_effect_color;
 
 	// If the camera is in a solid node, make everything black.
-	// (first person mode only)
-	if (features.solidness == 2 && cam_mode == CAMERA_MODE_FIRST &&
-		!m_control.allow_noclip) {
+	if (any_nearby_solid && !m_control.allow_noclip) {
```

Comment:
Was there any reason for this being first person mode only? Surely the 3rd person mode isn't implemented *that hackily*, placing the camera inside solid nodes and hoping backface culling does all the work? Please test this extensively in 3rd person mode.

Reply:
If this correctly uses the position of the camera (not player) in 3rd person mode I don't see any issue with it.
Currently it's not too hard to get the 3P mode to look inside nodes like in the OP screenshot.

Reply:
I have to make this first-person only unfortunately because otherwise underground the screen is black way too often (e.g. when you're in a tunnel or the like). This pretty much defeats the purpose of the PR.